crits/core/views.py

```diff
@@ -1390,9 +1390,19 @@ def user_context(request):
     """
 
     context = {}
-    # Get user theme
-    user = CRITsUser.objects(username=request.user.username).first()
-    if user:
+    user = request.user
+    if not user.username:
+        # Get user theme
+        user = CRITsUser.objects(username=request.user.username).first()
+        if user:
```

Comment:
Why is this entire block of code duplicated? There should be a better way to do this without duplicating the code or building another function. Seems like the condition checks here are off and should be done differently.